sqlglot/parser.py

```diff
@@ -6667,16 +6667,36 @@ def _parse_copy_parameters(self) -> t.List[exp.CopyParameter]:
 
         return options
 
+    def _parse_var_property(self) -> t.Optional[exp.Expression]:
+        prop = self._parse_property()
+
+        if not isinstance(prop, exp.Property):
+            return prop
+
+        # Replace the unquoted identifiers with vars
+        this = prop.this.find(exp.Identifier)
+        value = prop.args.get("value")
+        value = value.find(exp.Identifier) if value else None
+        if this and not this.quoted:
+            this.replace(exp.var(this.name))
+        if value and not value.quoted:
+            value.replace(exp.var(value.name))
```

Comment:
The RHS can vary from var, string literal, tuple of literals etc so it's not always as straightforward as parsing `<var> = <var>` afaict . With `parse_property` parsing the value/rhs as `_parse_bitwise` we catch all of these cases, with the exception that what should be parsed as var is now parsed as `exp.Column(exp.Identifier(...))`. 

For this reason, I thought it would be fine to reuse it and swap out these unquoted identifiers with `exp.Var`. In my upcoming commit I refactored this method completely and instead added an argument in `parse_property` that would force it to default to `exp.Var`